### ISSUE TITLE
run vgg_16 error

### ISSUE THREAD

**Issue by fbiswt** [baidu/Paddle#45]
I got a error when I run "train.sh" in demo/image_classification
I0907 14:35:07.593504 49407 Util.cpp:144] commandline: /home/hadoop/paddle/paddle/build/bin/../opt/paddle/bin/paddle_trainer --config=vgg_16_cifar.py --dot_period=10 --log_period=100 --test_all_data_in_one_period=1 --use_gpu=1 --trainer_count=1 --num_passes=200 --save_dir=./cifar_vgg_model 
I0907 14:35:08.002375 49407 Util.cpp:113] Calling runInitFunctions
I0907 14:35:08.002609 49407 Util.cpp:126] Call runInitFunctions done.
[INFO 2016-09-07 14:35:08,043 layers.py:1438] channels=3 size=3072
[INFO 2016-09-07 14:35:08,043 layers.py:1438] output size for __conv_0__ is 32 
[INFO 2016-09-07 14:35:08,044 layers.py:1438] channels=64 size=65536
[INFO 2016-09-07 14:35:08,045 layers.py:1438] output size for __conv_1__ is 32 
[INFO 2016-09-07 14:35:08,046 layers.py:1499] output size for __pool_0__ is 16*16 
[INFO 2016-09-07 14:35:08,046 layers.py:1438] channels=64 size=16384
[INFO 2016-09-07 14:35:08,046 layers.py:1438] output size for __conv_2__ is 16 
[INFO 2016-09-07 14:35:08,047 layers.py:1438] channels=128 size=32768
[INFO 2016-09-07 14:35:08,048 layers.py:1438] output size for __conv_3__ is 16 
[INFO 2016-09-07 14:35:08,049 layers.py:1499] output size for __pool_1__ is 8*8 
[INFO 2016-09-07 14:35:08,049 layers.py:1438] channels=128 size=8192
[INFO 2016-09-07 14:35:08,049 layers.py:1438] output size for __conv_4__ is 8 
[INFO 2016-09-07 14:35:08,051 layers.py:1438] channels=256 size=16384
[INFO 2016-09-07 14:35:08,051 layers.py:1438] output size for __conv_5__ is 8 
[INFO 2016-09-07 14:35:08,052 layers.py:1438] channels=256 size=16384
[INFO 2016-09-07 14:35:08,052 layers.py:1438] output size for __conv_6__ is 8 
[INFO 2016-09-07 14:35:08,053 layers.py:1499] output size for __pool_2__ is 4*4 
[INFO 2016-09-07 14:35:08,054 layers.py:1438] channels=256 size=4096
[INFO 2016-09-07 14:35:08,054 layers.py:1438] output size for __conv_7__ is 4 
[INFO 2016-09-07 14:35:08,055 layers.py:1438] channels=512 size=8192
[INFO 2016-09-07 14:35:08,055 layers.py:1438] output size for __conv_8__ is 4 
[INFO 2016-09-07 14:35:08,056 layers.py:1438] channels=512 size=8192
[INFO 2016-09-07 14:35:08,056 layers.py:1438] output size for __conv_9__ is 4 
[INFO 2016-09-07 14:35:08,058 layers.py:1499] output size for __pool_3__ is 2*2 
[INFO 2016-09-07 14:35:08,058 layers.py:1499] output size for __pool_4__ is 1*1 
[INFO 2016-09-07 14:35:08,060 networks.py:1122] The input order is [image, label]
[INFO 2016-09-07 14:35:08,060 networks.py:1129] The output order is [__cost_0__]
I0907 14:35:08.067443 49407 Trainer.cpp:169] trainer mode: Normal
I0907 14:35:08.075389 49407 PyDataProvider2.cpp:219] loading dataprovider image_provider::processData
[INFO 2016-09-07 14:35:08,109 image_provider.py:52] Image size: 32
[INFO 2016-09-07 14:35:08,109 image_provider.py:53] Meta path: data/cifar-out/batches/batches.meta
[INFO 2016-09-07 14:35:08,109 image_provider.py:58] DataProvider Initialization finished
I0907 14:35:08.109668 49407 PyDataProvider2.cpp:219] loading dataprovider image_provider::processData
[INFO 2016-09-07 14:35:08,109 image_provider.py:52] Image size: 32
[INFO 2016-09-07 14:35:08,109 image_provider.py:53] Meta path: data/cifar-out/batches/batches.meta
[INFO 2016-09-07 14:35:08,109 image_provider.py:58] DataProvider Initialization finished
I0907 14:35:08.109978 49407 GradientMachine.cpp:134] Initing parameters..
I0907 14:35:08.554066 49407 GradientMachine.cpp:141] Init parameters done.
Current Layer forward/backward stack is 
LayerName: __batch_norm_10__
LayerName: __fc_layer_0__
LayerName: __dropout_0__
LayerName: __pool_4__
LayerName: __pool_3__
LayerName: __batch_norm_9__
LayerName: __conv_9__
LayerName: __batch_norm_8__
LayerName: __conv_8__
LayerName: __batch_norm_7__
LayerName: __conv_7__
LayerName: __pool_2__
LayerName: __batch_norm_6__
LayerName: __conv_6__
LayerName: __batch_norm_5__
LayerName: __conv_5__
LayerName: __batch_norm_4__
LayerName: __conv_4__
LayerName: __pool_1__
LayerName: __batch_norm_3__
LayerName: __conv_3__
LayerName: __batch_norm_2__
LayerName: __conv_2__
LayerName: __pool_0__
LayerName: __batch_norm_1__
LayerName: __conv_1__
LayerName: __batch_norm_0__
LayerName: __conv_0__
LayerName: image
*** Aborted at 1473230129 (unix time) try "date -d @1473230129" if you are using GNU date ***
Current Layer forward/backward stack is 
PC: @     0x7fb72227a855 (unknown)
Current Layer forward/backward stack is 
*** SIGSEGV (@0x130701aa00) received by PID 49407 (TID 0x7fb7386fe800) from PID 117549568; stack trace: ***
Current Layer forward/backward stack is 
    @     0x7fb737fdf100 (unknown)
Current Layer forward/backward stack is 
    @     0x7fb72227a855 (unknown)
Current Layer forward/backward stack is 
    @           0x8b3350 hl_batch_norm_backward()
Current Layer forward/backward stack is 
    @           0x5d4684 paddle::CudnnBatchNormLayer::backward()
Current Layer forward/backward stack is 
    @           0x620bae paddle::NeuralNetwork::backward()
Current Layer forward/backward stack is 
    @           0x69c95d paddle::TrainerInternal::forwardBackwardBatch()
Current Layer forward/backward stack is 
    @           0x69cf14 paddle::TrainerInternal::trainOneBatch()
Current Layer forward/backward stack is 
    @           0x698350 paddle::Trainer::trainOnePass()
Current Layer forward/backward stack is 
    @           0x69ba47 paddle::Trainer::train()
Current Layer forward/backward stack is 
    @           0x53aea3 main
Current Layer forward/backward stack is 
    @     0x7fb73587bb15 __libc_start_main
Current Layer forward/backward stack is 
    @           0x545b15 (unknown)
/home/hadoop/paddle/paddle/build/bin/paddle: line 46: 49407 Segmentation fault      (core dumped) ${DEBUGGER} $MYDIR/../opt/paddle/bin/paddle_trainer ${@:2}
No data to plot. Exiting!

Someone know why?

**Comment by qingqing01** [comment]
@fbiswt please give me the version of your cuda and cudnn. Thanks.

**Comment by fbiswt** [comment]
@qingqing01 cuda 7.5 ,cudnn v4 . Many Thanks

**Comment by fbiswt** [comment]
@qingqing01  Should I change the version of cudnn to V5?

**Comment by qingqing01** [comment]
@fbiswt  It is ok when I tested with cuda 7.5, cudnn v4 or cudnn v5. But my GPU is Tesla K40, what is your GPU type?

**Comment by fbiswt** [comment]
@qingqing01 My GPU is also K40, but I install paddle on centos, is this problem?

**Comment by fbiswt** [comment]
@qingqing01 When I make paddle ,I got a problem is 
CMake Error at cmake/cudnn.cmake:33 (get_filename_component):
  get_filename_component unknown component DIRECTORY
Call Stack (most recent call first):
  CMakeLists.txt:51 (include)
so I delete the line "get_filename_component(CUDNN_LIB_PATH ${CUDNN_LIBRARY} DIRECTORY)
" in cudnn.make

**Comment by fbiswt** [comment]
@qingqing01 I upgrade cmake to cmake3 and solved "get_filename_component unknown component DIRECTORY" error ,and reinstall paddle, and also got that problem

**Comment by fbiswt** [comment]
Anyone answer this question ?

**Comment by qingqing01** [comment]
@fbiswt I'm sorry for this question and can't give you a solution now.  In fact, we have two implementations for BatchNorm. Maybe you can try another implementation by set batch_norm_type="batch_norm" in batch_norm_layer. A simple way, you can set batch_norm_type="batch_norm" in `python/paddle/trainer_config_helpers/layers.py` and reinstall paddle.

**Comment by fbiswt** [comment]
@qingqing01 Thank you so much for helping me solve this problem ! Now the paddle runing without error!

**Comment by aaronzs** [comment]
@qingqing01 Thank you! 
I can confirm the same problem happens on Fedora with CUDA 7.5, cuDNN 4, Tesla K80 GPU, and `batch_norm_type="batch_norm"` works.

**Comment by hedaoyuan** [comment]
@aaronzs & @fbiswt 
We check the code, and the bug is paddle use the CUDNN V4 RC not CUDNN V4. 
The reason for this bug is that the cudnnBatchNormalizationBackward API is different between the two versions.
We will fix this bug soon.
Before this use cudnn 5.0 is better.

**Comment by qingqing01** [comment]
We have fixed this problem, the code is #71
